### PR TITLE
[Logs] Sync up logger provider build-up scenarios with tracer provider

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

This PR syncs up the LoggerProvider scenarios (added on #3504) with the TracerProvider scenarios (#3533)

* AddExporter methods added
* AddOpenTelemetry is now safe to be called multiple times
* Removed the ability to register `BaseProcessor<LogRecord>`s & `Action<IServiceProvider, OpenTelemetryLoggerProvider>`s into the `IServiceCollection`
 
## Public API

```diff
namespace OpenTelemetry.Logs
{
   public class OpenTelemetryLoggerOptions
   {
+     public OpenTelemetryLoggerOptions AddExporter(ExportProcessorType exportProcessorType, BaseExporter<LogRecord> exporter) {}
+     public OpenTelemetryLoggerOptions AddExporter(ExportProcessorType exportProcessorType, BaseExporter<LogRecord> exporter, Action<ExportLogRecordProcessorOptions> configure) {}
+     public OpenTelemetryLoggerOptions AddExporter<T>(ExportProcessorType exportProcessorType) where T : BaseExporter<LogRecord> {}
+     public OpenTelemetryLoggerOptions AddExporter<T>(ExportProcessorType exportProcessorType, Action<ExportLogRecordProcessorOptions> configure) where T : BaseExporter<LogRecord> {}
   }

    // This class is similar to MetricReaderOptions and is used by AddExporter extensions
+   public class ExportLogRecordProcessorOptions
+   {
+      public ExportProcessorType ExportProcessorType { get; set; }
+      public BatchExportLogRecordProcessorOptions BatchExportProcessorOptions { get; set; }
+   }

    // This class is similar to BatchExportActivityProcessorOptions and exists so separate environment variables may be used to configure batch log processor settings
+   public class BatchExportLogRecordProcessorOptions
+   {
+   }
}
```

## TODOs

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Changes in public API reviewed
* [X] Unit tests

### CODE REVIEW

File: src/OpenTelemetry/Logs/BatchExportLogRecordProcessorOptions.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    /// </remarks>
+    public class BatchExportLogRecordProcessorOptions : BatchExportProcessorOptions<LogRecord>
+    {
+        internal const string MaxQueueSizeEnvVarKey = "OTEL_BLP_MAX_QUEUE_SIZE";
+
+        internal const string MaxExportBatchSizeEnvVarKey = "OTEL_BLP_MAX_EXPORT_BATCH_SIZE";
+
+        internal const string ExporterTimeoutEnvVarKey = "OTEL_BLP_EXPORT_TIMEOUT";
+
+        internal const string ScheduledDelayEnvVarKey = "OTEL_BLP_SCHEDULE_DELAY";
```
[the remark this answers is not in the document]
Comment (CodeBlanch) [reply src/OpenTelemetry/Logs/BatchExportLogRecordProcessorOptions.cs:48]: @pellared Not yet. I can spin one up, but would prefer not to tie this PR to something in spec which can take a long long time to get done. I do have a note above that they are experimental/subject to change. How you feel about that?

[the remark this answers is not in the document]
Comment (Kielek) [reply src/OpenTelemetry/Logs/BatchExportLogRecordProcessorOptions.cs:48]: https://github.com/open-telemetry/opentelemetry-specification/blob/6ce62202e5407518e19c56c445c13682ef51a51d/specification/sdk-environment-variables.md#language-specific-environment-variables

Maybe with this prefix `OTEL_DOTNET_BLP*` and mark as experimental?

[the remark this answers is not in the document]
Comment (CodeBlanch) [reply src/OpenTelemetry/Logs/BatchExportLogRecordProcessorOptions.cs]: > Maybe with this prefix OTEL_DOTNET_BLP*

Updated

> and mark as experimental?

I have the comment, is there something else you had in mind to denote that? Not sure of anything built-in we could use.

[the remark this answers is not in the document]
Comment (Kielek) [reply src/OpenTelemetry/Logs/BatchExportLogRecordProcessorOptions.cs]: > I have the comment, is there something else you had in mind to denote that? Not sure of anything built-in we could use.

I suppose that you will document it later, at least in changelog or package readme . I suppose that it is good place to clarify that it is temporary solution and could be changed when otel specification will be updated.

File: src/OpenTelemetry/Logs/BatchExportLogRecordProcessorOptions.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    /// <item>The environment variable keys are currently experimental and
+    /// subject to change. See: <see
+    /// href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#opentelemetry-environment-variable-specification">OpenTelemetry
+    /// Environment Variable Specification</see>.
+    /// </item>
+    /// </list>
+    /// </remarks>
+    public class BatchExportLogRecordProcessorOptions : BatchExportProcessorOptions<LogRecord>
+    {
+        internal const string MaxQueueSizeEnvVarKey = "OTEL_DOTNET_BLP_MAX_QUEUE_SIZE";
```
Comment (alanwest) [inline src/OpenTelemetry/Logs/BatchExportLogRecordProcessorOptions.cs:42]: I assume the spec will declare official environment variable names. You think we can be optimistic and remove `DOTNET` from these environment variable names and just go with `OTEL_BLP_MAX_QUEUE_SIZE` for example?

Comment (alanwest) [reply src/OpenTelemetry/Logs/BatchExportLogRecordProcessorOptions.cs:42]: oh scratch that i just saw the previous resolved conversation..

**Review by alanwest — APPROVED** [review]
LGTM

**Comment by CodeBlanch** [comment]
Merged to keep moving. Hoping to settle the environment variables before stable release.

**Pull request merged**